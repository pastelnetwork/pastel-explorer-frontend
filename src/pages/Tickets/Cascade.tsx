import Typography from '@material-ui/core/Typography';
import ArrowForwardIos from '@material-ui/icons/ArrowForwardIos';

import { Link } from '@components/Link/Link.styles';
import * as ROUTES from '@utils/constants/routes';
import { formatNumber } from '@utils/helpers/formatNumbers/formatNumbers';
import InfinityTable from '@components/InfinityTable/InfinityTable';
import { translate } from '@utils/helpers/i18n';

import { transformCascadeData, TTicketResponse } from './Tickets.helpers';
import { cascadeColumns } from './Tickets.columns';
import * as Styles from './Tickets.styles';

interface ICascadeProps {
  isMobile: boolean;
  ticketsData: TTicketResponse;
  innerWidth: number;
}

const Cascade: React.FC<ICascadeProps> = ({ isMobile, ticketsData, innerWidth }) => {
  const { data, total, isLoading, size, setSize } = ticketsData;

  const getTitle = () => {
    return (
      <Styles.BlockTitle className="latest-blocks">
        {translate('pages.tickets.cascadeTickets')} (
        {total > 1
          ? translate('pages.tickets.totalTickets', { total: formatNumber(total) })
          : translate('pages.tickets.totalTicket', { total: formatNumber(total) })}
        )
        <Link to={`${ROUTES.TICKETS_TYPE}/cascade`} className="view-all">
          <Typography align="center" className="p-16">
            {translate('pages.tickets.viewAll')} <ArrowForwardIos />
          </Typography>
        </Link>
      </Styles.BlockTitle>
    );
  };

  const handleFetchMoreMovements = (reachedTableBottom: boolean) => {
    if (!reachedTableBottom) return null;
    setSize(size + 1);
    return true;
  };

  const getRowHeight = () => {
    if (innerWidth < 600) {
      return 320;
    }

    if (isMobile) {
      return 180;
    }
    return 120;
  };

  return (
    <Styles.CascadeContainer id="cascadeTickets">
      <InfinityTable
        rows={data ? transformCascadeData(data) : []}
        columns={cascadeColumns}
        tableHeight={innerWidth < 600 ? 800 : 600}
        title={getTitle()}
        onBottomReach={handleFetchMoreMovements}
        className="data-table tickets-table"
        headerBackground
        customLoading={isLoading}
        rowHeight={getRowHeight()}
      />
    </Styles.CascadeContainer>
  );
};

export default Cascade;
