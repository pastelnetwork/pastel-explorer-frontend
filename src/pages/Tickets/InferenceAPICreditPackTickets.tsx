import { useState } from 'react';

import * as ROUTES from '@utils/constants/routes';
import InfinityTable from '@components/InfinityTable/InfinityTable';
import { translate } from '@utils/helpers/i18n';

import SectionTitle from './SectionTitle';
import {
  transformInferenceAPICreditPackData,
  TTicketResponse,
  useShowLess,
} from './Tickets.helpers';
import { offerAndTransferTicketsColumns } from './Tickets.columns';
import * as Styles from './Tickets.styles';

interface IInferenceAPICreditPackTicketsProps {
  innerWidth: number;
  ticketsData: TTicketResponse;
}

const InferenceAPICreditPackTickets: React.FC<IInferenceAPICreditPackTicketsProps> = ({
  innerWidth,
  ticketsData,
}) => {
  const { data, total, isLoading, size, setSize } = ticketsData;
  const [showLess, setShowLess] = useState(false);
  useShowLess(setShowLess);

  const handleFetchMoreMovements = (reachedTableBottom: boolean) => {
    if (!reachedTableBottom) return null;
    setSize(size + 1);
    return true;
  };

  const getTitle = () => {
    return (
      <SectionTitle
        title={translate('pages.tickets.inferenceAPICreditPackTickets')}
        total={total}
        toggleContent={() => setShowLess(!showLess)}
        showMore={showLess}
        viewAllLink={`${ROUTES.TICKETS_TYPE}/inference-api`}
      />
    );
  };

  const getRowHeight = () => {
    if (innerWidth < 600) {
      return 450;
    }
    if (innerWidth < 960) {
      return 240;
    }
    return 140;
  };

  const gettableHeight = () => {
    if (innerWidth < 600) {
      return 1200;
    }
    return 600;
  };

  return (
    <Styles.OtherTicketContainer id="offerTicketsAndTransferTickets">
      <InfinityTable
        rows={data ? transformInferenceAPICreditPackData(data) : []}
        columns={offerAndTransferTicketsColumns}
        tableHeight={gettableHeight()}
        title={getTitle()}
        onBottomReach={handleFetchMoreMovements}
        className="data-table tickets-table"
        headerBackground
        customLoading={isLoading}
        rowHeight={getRowHeight()}
        showLess={showLess}
      />
    </Styles.OtherTicketContainer>
  );
};

export default InferenceAPICreditPackTickets;
