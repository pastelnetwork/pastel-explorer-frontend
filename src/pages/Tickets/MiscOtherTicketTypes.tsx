import Typography from '@material-ui/core/Typography';
import ArrowForwardIos from '@material-ui/icons/ArrowForwardIos';

import { Link } from '@components/Link/Link.styles';
import * as ROUTES from '@utils/constants/routes';
import { formatNumber } from '@utils/helpers/formatNumbers/formatNumbers';
import InfinityTable from '@components/InfinityTable/InfinityTable';
import { translate } from '@utils/helpers/i18n';

import { transformOtherData, TTicketResponse } from './Tickets.helpers';
import { otherTicketsColumns } from './Tickets.columns';
import * as Styles from './Tickets.styles';

interface IMiscOtherTicketTypesProps {
  isMobile: boolean;
  ticketsData: TTicketResponse;
}

const MiscOtherTicketTypes: React.FC<IMiscOtherTicketTypesProps> = ({ isMobile, ticketsData }) => {
  const { data, total, isLoading, size, setSize } = ticketsData;

  const handleFetchMoreMovements = (reachedTableBottom: boolean) => {
    if (!reachedTableBottom) return null;
    setSize(size + 1);
    return true;
  };

  const getTitle = () => {
    return (
      <Styles.BlockTitle className="latest-blocks">
        {translate('pages.tickets.miscOtherTicketTypes')} (
        {total > 1
          ? translate('pages.tickets.totalTickets', { total: formatNumber(total) })
          : translate('pages.tickets.totalTicket', { total: formatNumber(total) })}
        )
        <Link to={`${ROUTES.TICKETS_TYPE}/other`} className="view-all">
          <Typography align="center" className="p-16">
            {translate('pages.tickets.viewAll')} <ArrowForwardIos />
          </Typography>
        </Link>
      </Styles.BlockTitle>
    );
  };

  return (
    <Styles.OtherTicketContainer id="miscOtherTicketTypes">
      <InfinityTable
        rows={data ? transformOtherData(data) : []}
        columns={otherTicketsColumns}
        tableHeight={495}
        title={getTitle()}
        onBottomReach={handleFetchMoreMovements}
        className="data-table"
        headerBackground
        customLoading={isLoading}
        rowHeight={isMobile ? 140 : 50}
      />
    </Styles.OtherTicketContainer>
  );
};

export default MiscOtherTicketTypes;
