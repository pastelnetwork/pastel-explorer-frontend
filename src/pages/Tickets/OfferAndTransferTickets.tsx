import Typography from '@material-ui/core/Typography';
import ArrowForwardIos from '@material-ui/icons/ArrowForwardIos';

import { Link } from '@components/Link/Link.styles';
import * as ROUTES from '@utils/constants/routes';
import { formatNumber } from '@utils/helpers/formatNumbers/formatNumbers';
import InfinityTable from '@components/InfinityTable/InfinityTable';
import { translate } from '@utils/helpers/i18n';

import { transformOfferAndTransferData, TTicketResponse } from './Tickets.helpers';
import { offerAndTransferTicketsColumns } from './Tickets.columns';
import * as Styles from './Tickets.styles';

interface IOfferAndTransferTicketsProps {
  innerWidth: number;
  ticketsData: TTicketResponse;
}

const OfferAndTransferTickets: React.FC<IOfferAndTransferTicketsProps> = ({
  innerWidth,
  ticketsData,
}) => {
  const { data, total, isLoading, size, setSize } = ticketsData;

  const handleFetchMoreMovements = (reachedTableBottom: boolean) => {
    if (!reachedTableBottom) return null;
    setSize(size + 1);
    return true;
  };

  const getTitle = () => {
    return (
      <Styles.BlockTitle className="latest-blocks">
        {translate('pages.tickets.offerTicketsAndTransferTickets')} (
        {total > 1
          ? translate('pages.tickets.totalTickets', { total: formatNumber(total) })
          : translate('pages.tickets.totalTicket', { total: formatNumber(total) })}
        )
        <Link to={`${ROUTES.TICKETS_TYPE}/offer-transfer`} className="view-all">
          <Typography align="center" className="p-16">
            {translate('pages.tickets.viewAll')} <ArrowForwardIos />
          </Typography>
        </Link>
      </Styles.BlockTitle>
    );
  };

  const getRowHeight = () => {
    if (innerWidth < 600) {
      return 450;
    }
    if (innerWidth < 960) {
      return 240;
    }
    return 140;
  };

  const gettableHeight = () => {
    if (innerWidth < 600) {
      return 1200;
    }
    return 600;
  };

  return (
    <Styles.OtherTicketContainer id="offerTicketsAndTransferTickets">
      <InfinityTable
        rows={data ? transformOfferAndTransferData(data) : []}
        columns={offerAndTransferTicketsColumns}
        tableHeight={gettableHeight()}
        title={getTitle()}
        onBottomReach={handleFetchMoreMovements}
        className="data-table tickets-table"
        headerBackground
        customLoading={isLoading}
        rowHeight={getRowHeight()}
      />
    </Styles.OtherTicketContainer>
  );
};

export default OfferAndTransferTickets;
