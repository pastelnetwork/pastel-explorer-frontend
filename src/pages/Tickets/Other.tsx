import Typography from '@material-ui/core/Typography';
import ArrowForwardIos from '@material-ui/icons/ArrowForwardIos';

import { Link } from '@components/Link/Link.styles';
import * as ROUTES from '@utils/constants/routes';
import { formatNumber } from '@utils/helpers/formatNumbers/formatNumbers';
import useTickets from '@hooks/useTickets';
import InfinityTable from '@components/InfinityTable/InfinityTable';
import { translate } from '@utils/helpers/i18n';

import { otherColumns, transformOtherData, DATA_LIMIT } from './Tickets.helpers';
import * as Styles from './Tickets.styles';

interface IOtherProps {
  isMobile: boolean;
}

const Other: React.FC<IOtherProps> = ({ isMobile }) => {
  const { data, total, isLoading, size, setSize } = useTickets('other', DATA_LIMIT);

  const handleFetchMoreMovements = (reachedTableBottom: boolean) => {
    if (!reachedTableBottom) return null;
    setSize(size + 1);
    return true;
  };

  const getTitle = () => {
    return (
      <Styles.BlockTitle className="latest-blocks">
        {translate('pages.tickets.otherTickets')} (
        {total > 1
          ? translate('pages.tickets.totalTickets', { total: formatNumber(total) })
          : translate('pages.tickets.totalTicket', { total: formatNumber(total) })}
        )
        <Link to={`${ROUTES.TICKETS_TYPE}/other`} className="view-all">
          <Typography align="center" className="p-16">
            {translate('pages.tickets.viewAll')} <ArrowForwardIos />
          </Typography>
        </Link>
      </Styles.BlockTitle>
    );
  };

  return (
    <Styles.OtherTicketContainer>
      <InfinityTable
        rows={data ? transformOtherData(data) : []}
        columns={otherColumns}
        tableHeight={495}
        title={getTitle()}
        onBottomReach={handleFetchMoreMovements}
        className="data-table"
        headerBackground
        customLoading={isLoading}
        rowHeight={isMobile ? 140 : 45}
      />
    </Styles.OtherTicketContainer>
  );
};

export default Other;
