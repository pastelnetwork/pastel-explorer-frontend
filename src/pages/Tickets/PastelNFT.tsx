import Typography from '@material-ui/core/Typography';
import ArrowForwardIos from '@material-ui/icons/ArrowForwardIos';

import { Link } from '@components/Link/Link.styles';
import * as ROUTES from '@utils/constants/routes';
import { formatNumber } from '@utils/helpers/formatNumbers/formatNumbers';
import useTickets from '@hooks/useTickets';
import InfinityTable from '@components/InfinityTable/InfinityTable';
import { translate } from '@utils/helpers/i18n';

import { pastelIdColumns, transformPastelIdData, DATA_LIMIT } from './Tickets.helpers';
import * as Styles from './Tickets.styles';

interface IPastelNFTProps {
  isMobile: boolean;
}

const PastelNFT: React.FC<IPastelNFTProps> = ({ isMobile }) => {
  const { data, total, isLoading, size, setSize } = useTickets('pastelid', DATA_LIMIT);

  const handleFetchMoreMovements = (reachedTableBottom: boolean) => {
    if (!reachedTableBottom) return null;
    setSize(size + 1);
    return true;
  };

  const getTitle = () => {
    return (
      <Styles.BlockTitle className="latest-blocks">
        {translate('pages.tickets.pastelNFTTickets')} (
        {total > 1
          ? translate('pages.tickets.totalTickets', { total: formatNumber(total) })
          : translate('pages.tickets.totalTicket', { total: formatNumber(total) })}
        )
        <Link to={`${ROUTES.TICKETS_TYPE}/pastelid`} className="view-all">
          <Typography align="center" className="p-16">
            {translate('pages.tickets.viewAll')} <ArrowForwardIos />
          </Typography>
        </Link>
      </Styles.BlockTitle>
    );
  };

  return (
    <Styles.PastelContainer>
      <InfinityTable
        rows={data ? transformPastelIdData(data) : []}
        columns={pastelIdColumns}
        tableHeight={495}
        title={getTitle()}
        onBottomReach={handleFetchMoreMovements}
        className="data-table"
        headerBackground
        customLoading={isLoading}
        rowHeight={isMobile ? 140 : 45}
      />
    </Styles.PastelContainer>
  );
};

export default PastelNFT;
