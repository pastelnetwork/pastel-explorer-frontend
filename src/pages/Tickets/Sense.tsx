import Typography from '@material-ui/core/Typography';
import ArrowForwardIos from '@material-ui/icons/ArrowForwardIos';
import Skeleton from '@material-ui/lab/Skeleton';
import Paper from '@material-ui/core/Paper';
import TableContainer from '@material-ui/core/TableContainer';
import Table from '@material-ui/core/Table';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableBody from '@material-ui/core/TableBody';

import { Link } from '@components/Link/Link.styles';
import RouterLink from '@components/RouterLink/RouterLink';
import * as ROUTES from '@utils/constants/routes';
import { getCurrencyName } from '@utils/appInfo';
import { formatAddress } from '@utils/helpers/format';
import { formatFullDate } from '@utils/helpers/date/date';
import { formatNumber } from '@utils/helpers/formatNumbers/formatNumbers';
import useTickets from '@hooks/useTickets';
import * as ExplorerStyles from '@pages/Explorer/Explorer.styles';

import { StyledTableCell, StyledTableRow } from './Tickets.helpers';

import * as Styles from './Tickets.styles';

const Sense: React.FC = () => {
  const { data, total, isLoading } = useTickets('sense', 10);
  return (
    <Styles.SenseContainer>
      <ExplorerStyles.BlockWrapper className="mt-24 latest-transactions-wrapper">
        <ExplorerStyles.BlockTitle className="latest-blocks">
          Sense tickets (Total {formatNumber(total)} tickets)
          <Link to={`${ROUTES.TICKETS_TYPE}/sense`} className="view-all">
            <Typography align="center" className="p-16">
              View all <ArrowForwardIos />
            </Typography>
          </Link>
        </ExplorerStyles.BlockTitle>
        <TableContainer component={Paper} className="table-container">
          <Table aria-label="customized table" className="custom-table latest-transactions">
            <TableHead className="table__row-header">
              <TableRow>
                <StyledTableCell>Block</StyledTableCell>
                <StyledTableCell>TXID</StyledTableCell>
                <StyledTableCell>Image Hash</StyledTableCell>
                <StyledTableCell>Fee</StyledTableCell>
                <StyledTableCell>Timestamp</StyledTableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {data.map(item => (
                <StyledTableRow key={item.transactionHash} className="table__row">
                  <StyledTableCell component="td" scope="row">
                    <RouterLink
                      route={`${ROUTES.BLOCK_DETAILS}/${item.height}`}
                      value={item.height}
                      title={item.height.toString()}
                      className="address-link"
                    />
                  </StyledTableCell>
                  <StyledTableCell component="td" scope="row">
                    <RouterLink
                      route={`${ROUTES.TRANSACTION_DETAILS}/${item.transactionHash}`}
                      value={formatAddress(item.transactionHash, 12)}
                      title={item.transactionHash}
                      className="address-link"
                    />
                  </StyledTableCell>
                  <StyledTableCell>
                    {item.imageHash ? (
                      <RouterLink
                        route={`${ROUTES.SENSE_DETAILS}/${item.transactionHash}/${item.imageHash}`}
                        value={formatAddress(item.imageHash, 12)}
                        title={item.imageHash}
                        className="address-link"
                      />
                    ) : (
                      '--'
                    )}
                  </StyledTableCell>
                  <StyledTableCell className="nowrap">
                    {item.fee} {getCurrencyName()}
                  </StyledTableCell>
                  <StyledTableCell className="nowrap">
                    {item.timestamp ? formatFullDate(item.timestamp) : '--'}
                  </StyledTableCell>
                </StyledTableRow>
              ))}
              {!data.length && !isLoading ? (
                <StyledTableRow>
                  <StyledTableCell component="td" scope="row" colSpan={5} style={{ padding: 0 }}>
                    <Typography className="empty-label">Data not found</Typography>
                  </StyledTableCell>
                </StyledTableRow>
              ) : null}
              {isLoading ? (
                <StyledTableRow>
                  <StyledTableCell component="td" scope="row" colSpan={6} style={{ padding: 0 }}>
                    <Skeleton animation="wave" variant="rect" height={150} />
                  </StyledTableCell>
                </StyledTableRow>
              ) : null}
            </TableBody>
          </Table>
        </TableContainer>
      </ExplorerStyles.BlockWrapper>
    </Styles.SenseContainer>
  );
};

export default Sense;
