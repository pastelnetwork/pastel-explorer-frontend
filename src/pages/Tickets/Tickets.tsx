import { useState, useEffect } from 'react';
import { Grid } from '@material-ui/core';
import { Skeleton } from '@material-ui/lab';

import { useUsdPrice } from '@hooks/useTransactionDetails';
import useTickets from '@hooks/useTickets';

import Sense from './Sense';
import Cascade from './Cascade';
import PastelIDAndUsernameTickets from './PastelIDAndUsernameTickets';
import PastelNftTickets from './PastelNftTickets';
import OfferAndTransferTickets from './OfferAndTransferTickets';
import MiscOtherTicketTypes from './MiscOtherTicketTypes';
import {
  ticketsSummary,
  DATA_LIMIT,
  getTotalTickets,
  TTicketResponse,
  isLoading,
} from './Tickets.helpers';
import * as Styles from './Tickets.styles';

const Tickets: React.FC = () => {
  const { usdPrice } = useUsdPrice();
  const otherTicketData = useTickets('other', DATA_LIMIT);
  const cascadeTicketData = useTickets('cascade', DATA_LIMIT);
  const offerTransferTicketData = useTickets('offer-transfer', DATA_LIMIT);
  const pastelidUsenameTicketData = useTickets('pastelid-usename', DATA_LIMIT);
  const pastelNftTicketData = useTickets('pastel-nft', DATA_LIMIT);
  const senseTicketData = useTickets('sense', DATA_LIMIT);
  const [isMobile, setMobileView] = useState(false);
  const [innerWidth, setInnerWidth] = useState(0);

  const handleResize = () => {
    setMobileView(false);
    setInnerWidth(window.innerWidth);
    if (window.innerWidth < 960) {
      setMobileView(true);
    }
  };

  useEffect(() => {
    handleResize();

    window.addEventListener('resize', handleResize);
    return () => {
      window.removeEventListener('resize', handleResize);
    };
  }, []);

  const ticketData = {
    senseTickets: senseTicketData.total,
    cascadeTickets: cascadeTicketData.total,
    pastelIDAndUsernameTickets: pastelidUsenameTicketData.total,
    pastelNFTTickets: pastelNftTicketData.total,
    offerTicketsAndTransferTickets: offerTransferTicketData.total,
    miscOtherTicketTypes: otherTicketData.total,
  };
  const ticketLoading = {
    senseTickets: senseTicketData.isLoading,
    cascadeTickets: cascadeTicketData.isLoading,
    pastelIDAndUsernameTickets: pastelidUsenameTicketData.isLoading,
    pastelNFTTickets: pastelNftTicketData.isLoading,
    offerTicketsAndTransferTickets: offerTransferTicketData.isLoading,
    miscOtherTicketTypes: otherTicketData.isLoading,
  };

  return (
    <Styles.TicketsContainer>
      <Grid container spacing={6}>
        <Styles.GirdStyle item className="full">
          <Styles.TicketSummaryContainer>
            {ticketsSummary.map(item => (
              <Styles.TicketSummaryBox key={item.id} to={item.link} className={item.id}>
                <span className="ticket-summary-title">{item.name}</span>
                <span className="ticket-summary-value">
                  {isLoading(item.id, ticketLoading) ? (
                    <Skeleton animation="wave" variant="text" />
                  ) : (
                    getTotalTickets(item.id, ticketData)
                  )}
                </span>
              </Styles.TicketSummaryBox>
            ))}
          </Styles.TicketSummaryContainer>
        </Styles.GirdStyle>
      </Grid>
      <Grid container spacing={6}>
        <Styles.GirdStyle item xs={12} lg={6} className="left">
          <Sense
            isMobile={isMobile}
            ticketsData={senseTicketData as TTicketResponse}
            innerWidth={innerWidth}
            usdPrice={usdPrice}
          />
        </Styles.GirdStyle>
        <Styles.GirdStyle item xs={12} lg={6} className="right">
          <Cascade
            isMobile={isMobile}
            ticketsData={cascadeTicketData as TTicketResponse}
            innerWidth={innerWidth}
            usdPrice={usdPrice}
          />
        </Styles.GirdStyle>
      </Grid>
      <Grid container spacing={6}>
        <Styles.GirdStyle item xs={12} lg={6} className="left">
          <PastelIDAndUsernameTickets
            isMobile={isMobile}
            ticketsData={pastelidUsenameTicketData as TTicketResponse}
          />
        </Styles.GirdStyle>
        <Styles.GirdStyle item xs={12} lg={6} className="right">
          <PastelNftTickets
            isMobile={isMobile}
            ticketsData={pastelNftTicketData as TTicketResponse}
            innerWidth={innerWidth}
          />
        </Styles.GirdStyle>
      </Grid>
      <Grid container spacing={6}>
        <Styles.GirdStyle item xs={12} lg={6} className="left">
          <OfferAndTransferTickets
            isMobile={isMobile}
            ticketsData={offerTransferTicketData as TTicketResponse}
          />
        </Styles.GirdStyle>
        <Styles.GirdStyle item xs={12} lg={6} className="right">
          <MiscOtherTicketTypes
            isMobile={isMobile}
            ticketsData={otherTicketData as TTicketResponse}
          />
        </Styles.GirdStyle>
      </Grid>
    </Styles.TicketsContainer>
  );
};

export default Tickets;
