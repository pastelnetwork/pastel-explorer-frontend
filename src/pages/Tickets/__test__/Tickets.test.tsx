import { shallow } from 'enzyme';
import { Grid } from '@material-ui/core';
import 'jest-styled-components';

import i18next from '../../../utils/helpers/i18n';
import Tickets from '../Tickets';
import Sense from '../Sense';
import Cascade from '../Cascade';
import PastelNFT from '../PastelNFT';
import Other from '../Other';
import * as Styles from '../Tickets.styles';

jest.mock('i18next-http-backend');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            // noop
          }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {
      // noop
    },
  },
}));
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
});

describe('pages/Tickets', () => {
  const wrapper = shallow(<Tickets />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <Grid>', () => {
    expect(wrapper.find(Grid).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Sense>', () => {
    expect(wrapper.find(Sense).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Cascade>', () => {
    expect(wrapper.find(Cascade).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <PastelNFT>', () => {
    expect(wrapper.find(PastelNFT).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Other>', () => {
    expect(wrapper.find(Other).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.TicketsContainer>', () => {
    expect(wrapper.find(Styles.TicketsContainer).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.GirdStyle>', () => {
    expect(wrapper.find(Styles.GirdStyle).length).toBeGreaterThanOrEqual(1);
  });
});
