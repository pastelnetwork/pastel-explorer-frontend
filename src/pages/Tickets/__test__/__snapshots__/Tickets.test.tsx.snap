// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Tickets renders correctly 1`] = `ShallowWrapper {}`;
